sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -114,7 +114,8 @@ case class AnalysisContext(
     nestedViewDepth: Int = 0,
     maxNestedViewDepth: Int = -1,
     relationCache: mutable.Map[Seq[String], LogicalPlan] = mutable.Map.empty,
-    referredTempViewNames: Seq[Seq[String]] = Seq.empty)
+    referredTempViewNames: Seq[Seq[String]] = Seq.empty,
+    isTempView: Boolean = false)
```

Comment:
then shall we just use `referredTempFuncNames`?

Reply:
if we change the type to referredTempFuncNames: Option[Seq[String]], then it can replace `isTempView`. Because Seq.empty can be permanent view or temp view refers non temp functions.

Reply:
yea let's do it then. We can use `isTempView: Boolean = false` in the backport PR of 3.0/2.4.

Reply:
you are correct in the first comment. we can just use `referredTempFunctionNames`. But this approach will fix both permanent and temp view problems, which I was planning to fix them separately. Anyway, I switch to use `referredTempFunctionNames` in the latest commit.

BTW, in 3.0/2.4, temp view stores analyzed plan, and won't be wrapped in `View`, so we don't need the `isTempView` flag.